Adding commands

KolbyMcGarrah opened the pull request:
Updated recordCall function in the observability.go file to except the redis.Nil error returned on a nil return. This fixes the issue of setting the status metric to ERROR when a nil value is returned from a get.

Added a Ping and Incr command to the Wrapper that allow us to use the wrapper to preform those needed actions. Ping returns a IntCmd, so I also implemented that struct in the commands.go file.

Commit:
Adding new commands and updating error check

added Ping, and Incr functionality to the wrapper. Updated recordCallFunc error check to look out for Nil.

Commit:
Updating Pip command to return a string with the error

otternq commented on wrapper.go:
```diff
@@ -40,3 +40,21 @@ func (w *Wrapper) Set(ctx context.Context, key string, value interface{}, expira
+
+}
+
+func (w *Wrapper) Ping() (error, string) {
```
I'm trying to ensure this wrapper returns the same types (well interface of that type) as the actual clients function calls. It looks like [Ping](https://godoc.org/gopkg.in/redis.v3#Client.Ping) returns a [StatusCmd](https://godoc.org/gopkg.in/redis.v3#StatusCmd) not `(error, string)`. If I'm not missing something I'd like this `Ping` to return the same.

otternq replied:
What do you think of supplying a `context.Context` to `Ping` and calling `recordCallFunc` ?

KolbyMcGarrah replied:
I'll update ping to return that StatusCmd. As for accepting the context, I also thought about that originally as well, but that Ping command would happen before a request is received so we'd have to send some other context with it which is also probably fine.

otternq replied:
Using a `context.Background()` would work if there isn't another context available yet

otternq reviewed (changes requested):
The updates look good

Commit:
Update Ping return vallue and metric call

Ensured that Ping returned the StatusCmd type that was being returned by the original call. Added context as a parameter to be able to record the Ping metrics.

Commit:
syncing variable name on the Ping function.
